components/net/cookie_storage.rs

```diff
@@ -65,7 +78,61 @@ impl CookieStorage {
         }
 
         // Step 12
-        self.cookies.push(cookie);
+        let domain = reg_host(&cookie.cookie.domain.clone().unwrap_or("".to_owned())).unwrap_or("".to_owned());
+        if !self.cookies_map.contains_key(&domain) {
+        let tempdomain = domain.clone();
+            let cookies: Vec<Cookie> = Vec::new();
+            self.cookies_map.insert(tempdomain, cookies);
+        }
+        let mut cookies = self.cookies_map.get_mut(&domain).unwrap();
+
+        if cookies.len() == self.max_per_host {
+            // Step 12.1
+            let old_len = cookies.len();
+            cookies.retain(|c| !check_cookie_expired(&c));
+            let new_len = cookies.len();
+
+            // https://datatracker.ietf.org/doc/draft-ietf-httpbis-cookie-alone
+            if new_len == old_len {
+                // Remove non-secure cookie with oldest access time
+                let mut is_first = true;
+                let mut index = 0;
+                let mut acc_time = time::now();
+                for i in 0..cookies.len() {
+                    let c = cookies.get(i).unwrap();
+                    if !c.cookie.secure && (is_first || c.last_access < acc_time) {
+                        acc_time = c.last_access;
+                        index = i;
+
+                        is_first = false;
+                    }
+                }
+
+                // All secure cookies were found
+                if is_first {
+                        if !cookie.cookie.secure {
+                            return;
+                        } else {
+                            let mut is_first = true;
+                            let mut index = 0;
+                            let mut acc_time = time::now();
+                            // Get secure cookie with the oldest access time
+                            for i in 0..cookies.len() {
```

Comment:
We should use the same pattern here as my previous review comment. We could even add a helper method that took a boolean argument to compare against the cookie's security to avoid the code duplication.